python/config_settings/transition.bzl

```diff
@@ -14,266 +14,15 @@
-    srcs = pyargs["srcs"]
-    deps = pyargs["deps"]
-    main = pyargs["main"]
-
-    # Attributes common to all build rules.
-    # https://bazel.build/reference/be/common-definitions#common-attributes
-    compatible_with = kwargs.pop("compatible_with", None)
-    deprecation = kwargs.pop("deprecation", None)
-    exec_compatible_with = kwargs.pop("exec_compatible_with", None)
-    exec_properties = kwargs.pop("exec_properties", None)
-    features = kwargs.pop("features", None)
-    restricted_to = kwargs.pop("restricted_to", None)
-    tags = kwargs.pop("tags", None)
-    target_compatible_with = kwargs.pop("target_compatible_with", None)
-    testonly = kwargs.pop("testonly", None)
-    toolchains = kwargs.pop("toolchains", None)
-    visibility = kwargs.pop("visibility", None)
-
-    common_attrs = {
-        "compatible_with": compatible_with,
-        "deprecation": deprecation,
-        "exec_compatible_with": exec_compatible_with,
-        "exec_properties": exec_properties,
-        "features": features,
-        "restricted_to": restricted_to,
-        "target_compatible_with": target_compatible_with,
-        "testonly": testonly,
-        "toolchains": toolchains,
-    }
-
-    # Test-specific extra attributes.
-    if "env_inherit" in kwargs:
-        common_attrs["env_inherit"] = kwargs.pop("env_inherit")
-    if "size" in kwargs:
-        common_attrs["size"] = kwargs.pop("size")
-    if "timeout" in kwargs:
-        common_attrs["timeout"] = kwargs.pop("timeout")
-    if "flaky" in kwargs:
-        common_attrs["flaky"] = kwargs.pop("flaky")
-    if "shard_count" in kwargs:
-        common_attrs["shard_count"] = kwargs.pop("shard_count")
-    if "local" in kwargs:
-        common_attrs["local"] = kwargs.pop("local")
-
-    # Binary-specific extra attributes.
-    if "output_licenses" in kwargs:
-        common_attrs["output_licenses"] = kwargs.pop("output_licenses")
-
-    rule_impl(
-        name = "_" + name,
-        args = args,
-        data = data,
-        deps = deps,
-        env = env,
-        srcs = srcs,
-        main = main,
-        tags = ["manual"] + (tags if tags else []),
-        visibility = ["//visibility:private"],
-        **dicts.add(common_attrs, kwargs)
-    )
-
-    return transition_rule(
-        name = name,
-        args = args,
-        deps = deps,
-        env = env,
-        python_version = python_version,
-        srcs = srcs,
-        tags = tags,
-        target = ":_" + name,
-        tools = data,
-        visibility = visibility,
-        **common_attrs
-    )
-
-def py_binary(name, python_version, **kwargs):
-    return _py_rule(_py_binary, _transition_py_binary, name, python_version, **kwargs)
-
-def py_test(name, python_version, **kwargs):
-    return _py_rule(_py_test, _transition_py_test, name, python_version, **kwargs)
 
-def _strip_suffix(s, suffix):
-    if s.endswith(suffix):
-        return s[:-len(suffix)]
-    else:
-        return s
+py_binary = _py_binary
```

Comment:
Very small nit, we could create a macro that populate the deprecation message, that way users dont need to read our chngelog, which they might not do.

Reply:
Good idea. Done